prow/flagutil/github.go

```diff
@@ -34,6 +35,8 @@ type GitHubOptions struct {
 	graphqlEndpoint     string
 	TokenPath           string
 	deprecatedTokenFile string
+	gitHost             string
+	useSSH              bool
```

Comment:
Aren't these fields specific to GitHub, not Git? I know we can create git clients from GitHub flags, but the opposite isn't true so we should be careful about combining these ideas.